js/xt.js

```diff
@@ -0,0 +1,1072 @@
+        //      19413.67,
+        //      3.679209
+        //      ],
+        //     ]
+        // }
+        //
+        // swap
+        //
+        // "result": {
+        //     "t": 1664279607831,
+        //     "s": "btc_usdt",
+        //     "u": 145028995606381060,
+        //     "b": [
+        //     [
+        //     "20281.5",
+        //     "48914"
+        //     ],
+        //     "a": [
+        //     "20281.5",
+        //     "48914"
+        //     ]
+        //     ]
+        // }
+        //
+        if (market['spot']) {
+            const request = {
+                'market': market['id'],
+            };
+            const response = await this.spotPublicGetDataApiV1GetDepth (this.extend (request, params));
+            const timestamp = this.safeString (response, 'time');
+            return this.parseOrderBook (response, symbol, timestamp);
+        } else if (market['swap']) {
+            const request = {
+                'symbol': market['id'],
+            };
+            if (limit !== undefined) {
+                request['level'] = Math.min (limit, 50);
+            } else {
+                request['level'] = 50;
+            }
+            let response = undefined;
+            if (market['linear']) {
+                response = await this.usdtPublicGetFutureMarketV1PublicQDepth (this.extend (request, params));
+            } else if (market['inverse']) {
+                response = await this.coinPublicGetFutureMarketV1PublicQDepth (this.extend (request, params));
+            }
+            const data = this.safeValue (response, 'result');
+            const timestamp = this.safeNumber (data, 't');
+            return this.parseOrderBook (data, symbol, timestamp, 'b', 'a');
+        }
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name xt#fetchTrades
+         * @description get the list of most recent trades for a particular symbol
+         * @param {string} symbol unified symbol of the market to fetch trades for
+         * @param {int|undefined} since timestamp in ms of the earliest trade to fetch
+         * @param {int|undefined} limit the maximum amount of trades to fetch
+         * @param {object} params extra parameters specific to the xt api endpoint
+         * @returns {[object]} a list of [trade structures]{@link https://docs.ccxt.com/en/latest/manual.html?#public-trades}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        const trades = await this.publicGetDataApiV1GetTrades (this.extend (request, params));
+        //
+        // [
+        //     [
+        //     1663887610712,
+        //     19391.17,
+        //     0.0032,
+        //     "bid",
+        //     "6978850460677441538"
+        //     ],
+        // ]
+        //
+        const result = [];
```

Comment:
As I mentioned earlier, we should follow the request here + delegate it to the parser (`parseTrades` in this case) that will call `parseTrade`... you know the drill